app/src/main/java/com/epfl/beatlink/repository/network/NetworkStatusTracker.kt

```diff
@@ -0,0 +1,64 @@
+package com.epfl.beatlink.repository.network
+
+import android.content.Context
+import android.net.ConnectivityManager
+import android.net.Network
+import android.net.NetworkCapabilities
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+
+/**
+ * Class that tracks the network status and provides it to the UI.
```

Comment:
It would be more accurate to say that this class tracks the network status and provides it to the ViewModel, which in turn communicates it to the UI.